### ISSUE TITLE
Type error: undefined is not a function

### ISSUE THREAD

**Issue by lancesims** [WickyNilliams/enquire.js#124]
While calling unregister('(max-width: 767px)') the undefined error is thrown.  Details are as follows:

Inspecting the enquire object in Chrome Dev Tools shows that the there are queries in the queries object, etc.
![screen shot 2014-12-03 at 4 51 32 pm](https://cloud.githubusercontent.com/assets/5647795/5290487/d8009510-7b0c-11e4-9382-215a15b392f4.png)

This happens during an unregister

On line 285 an error is thrown, and complete stack trace from my application is:

TypeError: undefined is not a function
    at Object.MediaQuery.clear (http://0.0.0.0:9000/bower_components/enquire/dist/enquire.js:203:22)
    at Object.MediaQueryDispatch.unregister (http://0.0.0.0:9000/bower_components/enquire/dist/enquire.js:285:27)
    at Object.unregisterEnquireBreakpoint (http://0.0.0.0:9000/components/breakpoint/breakpoint-service.js:63:37)
    at Object.$stateProvider.state.state.onExit (http://0.0.0.0:9000/app.js:66:39)
    at Object.invoke (http://0.0.0.0:9000/bower_components/angular/angular.js:4138:17)
    at $state.transitionTo.$state.transition.resolved.then.$state.transition (http://0.0.0.0:9000/bower_components/angular-ui-router/release/angular-ui-router.js:3173:23)
    at processQueue (http://0.0.0.0:9000/bower_components/angular/angular.js:13071:27)
    at http://0.0.0.0:9000/bower_components/angular/angular.js:13087:27
    at Scope.$get.Scope.$eval (http://0.0.0.0:9000/bower_components/angular/angular.js:14287:28)
    at Scope.$get.Scope.$digest (http://0.0.0.0:9000/bower_components/angular/angular.js:14103:31)

**Comment by Dechowmedia** [comment]
Please post your code / call to the unregister function.

Best of all make a test case of this on http://codepen.io or some other code hosts.

Then I'll try to make a fix.

**Comment by intrasoft-eu** [comment]
The same for me on Chrome 40. It's connected with this part:

```javascript
this.listener = function(mql) {
  self.mql = mql;
  self.assess();
};
```

My quick fix:

```javascript
this.listener = function(mql) {
  if (mql.currentTarget) {
    self.mql = mql.currentTarget;
  } else {
    self.mql = mql;
  }
  self.assess();
};
```

**Comment by PoeHaH** [comment]
I have the same problem when opening the print dialog in chrome (ctrl+p), then closing it again (click cancel) and continue navigating on the website. 

Whenever an unregister is happening, the debugger throws "undefined not a function", because 'mql' is not a MediaQueryList but a MediaQueryListEvent, which doesn't have the removeListener method, and thus the error is thrown.

**Comment by ionutzp** [comment]
any progress on this?

**Comment by rosszurowski** [comment]
I ran into the same issue with a different setup ([minimal test case](https://jsfiddle.net/p06kuw35/)). It seems like @intrasoft-eu's fix did the trick, so I've adapted it into a PR with the fix (#154). 

Since it seems like this library hasn't been updated in a while, I've temporarily published this fix to npm under a scoped to make it accessible:

```
npm install @rosszurowski/enquire --save
```

**Comment by WickyNilliams** [comment]
Sorry for the delay, all. This is now fixed in `v2.1.4` and is published to npm
